src/app/code/community/Omise/Gateway/Model/Payment/Base/Payment.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+
+class Omise_Gateway_Model_Payment_Base_Payment extends Omise_Gateway_Model_Payment {
+    /**
+     * Instantiate state and set it to state object
+     *
+     * @param string        $payment_action
+     * @param Varien_Object $state_object
+     */
+    public function initialize($payment_action, $state_object)
+    {
+        $title =  (string)Mage::getConfig()->getNode('default/payment/'.$this->getCode().'/title');
+        $payment = $this->getInfoInstance();
+        $order   = $payment->getOrder();
+
+        $invoice = $order->prepareInvoice();
+        $invoice->setIsPaid(false)->register();
+
+        $charge = $this->process($payment, $invoice->getBaseGrandTotal());
+
+        $payment->setCreatedInvoice($invoice)
+            ->setIsTransactionClosed(false)
+            ->setIsTransactionPending(true)
+            ->addTransaction(
+                Mage_Sales_Model_Order_Payment_Transaction::TYPE_ORDER,
+                $invoice,
+                false,
+                Mage::helper('omise_gateway')->__('Processing an amount of %s via Omise '.$title.' payment.', $order->getBaseCurrency()->formatTxt($invoice->getBaseGrandTotal()))
+            );
+
+        $order->addRelatedObject($invoice);
+
+        if ($charge->isAwaitPayment()) {
+            $state_object->setState(Mage_Sales_Model_Order::STATE_PENDING_PAYMENT);
+            $state_object->setStatus($order->getConfig()->getStateDefaultStatus(Mage_Sales_Model_Order::STATE_PENDING_PAYMENT));
+            $state_object->setIsNotified(false);
+
+            return;
+        }
+
+        $this->_suspectToBeFailed($payment);
+    }
+
+    /**
+     * Check method for processing with base currency
+     *
+     * @param  string $currencyCode
+     *
+     * @return boolean
+     */
+    public function canUseForCurrency($currencyCode)
+    {
+        if(isset($this->_currencies))
+            return (
+                in_array(strtoupper($currencyCode), $this->_currencies)
```

Comment:
Wondering if it might be an idea to put the currencies array items in lower case, as Magento seems to be using lower case for currency codes and it would remove the need for these `strtoupper` conversions